caas/kubernetes/provider/application/application.go

```diff
@@ -1026,31 +1033,97 @@ func (a *app) Delete() error {
 	// Cleanup lists of resources.
 	cleanup := []resources.Resource(nil)
 
+	// Create selector labels.
+	appLabel := utils.SelectorLabelsForApp(a.name, a.labelVersion)
+	modelLabel := utils.LabelsForModel(a.modelName, a.modelUUID, a.controllerUUID, a.labelVersion)
+	mergedLabel := utils.LabelsMerge(appLabel, modelLabel)
+
+	// List mutating webhook configurations to be deleted.
+	mws, err := resources.ListMutatingWebhookConfigs(ctx, a.client.AdmissionregistrationV1().MutatingWebhookConfigurations(), metav1.ListOptions{
```

Comment:
We're missing a unit test for checking that if an app creates all these resources, they are deleted.
The original podspec deletion logic also deleted any deployments or daemonsets that the app may have created, so we should include those as well
https://github.com/juju/juju/blob/83b35794fd57e81d54d7d6471f4f9543702addbc/caas/kubernetes/provider/k8s.go#L791